### PR TITLE
(1 / N) Add edit method to EditRequest, rename the update EditRequest to IncrementalChoiceMapRequest.

### PR DESCRIPTION
**Pull request by femtomc**

### CODE REVIEW

File: src/genjax/_src/core/generative/generative_function.py [hunk 1]
```diff
@@ -1519,3 +1516,17 @@ def assess(
...
+class IncrementalChoiceMapRequest(EditRequest):
+    constraint: ChoiceMapConstraint
+
+    def edit(
+        self,
+        key: PRNGKey,
+        tr: Trace[R],
+        argdiffs: Argdiffs,
+    ) -> tuple[Trace[R], Weight, Retdiff[R], "EditRequest"]:
+        gen_fn = tr.get_gen_fn()
```
Comment (MathieuHuot) [inline src/genjax/_src/core/generative/generative_function.py:1530]: add a test for this?

File: src/genjax/_src/core/generative/generative_function.py [hunk 2]
```diff
@@ -485,24 +484,24 @@ def model(var):
 
         An `EditRequest` denotes a function $tr \\mapsto (T, T')$ from traces to a pair of targets (the previous [`Target`][genjax.inference.Target] $T$, and the final [`Target`][genjax.inference.Target] $T'$).
 
-        Several common types of moves can be requested via the `IncrementalGenericRequest` type:
+        Several common types of moves can be requested via the `IncrementalChoiceMapRequest` type:
```
Comment (MathieuHuot) [inline src/genjax/_src/core/generative/generative_function.py:486]: I'm fine with it but why the name change?

File: src/genjax/_src/core/generative/generative_function.py (hunk 1 above)
Comment (femtomc) [reply src/genjax/_src/core/generative/generative_function.py:1530]: Did a bit of code swapping so this code path is hit by our current tests.

File: src/genjax/_src/core/generative/generative_function.py (hunk 2 above)
Comment (femtomc) [reply src/genjax/_src/core/generative/generative_function.py:486]: Make it easier to understand, I even went further in: https://github.com/probcomp/genjax/pull/1372/files#diff-cf05070a53a8ff04406eda8ff4d429024ccba01460279145af0a3ff56a6679e0

changing it to "ChoiceMapChange" (that's really what's going on)

**Review by MathieuHuot — APPROVED** [review]

**Pull request merged**